src/classes/ExpectSObject.cls

```diff
@@ -0,0 +1,72 @@
+public class ExpectSObject extends Expect {
+    private final SObject expected;
+
+    public ExpectSObject(SObject expected) {
+        this.expected = expected;
+    }
+
+    protected override String getExpectedAsString() {
+        return String.valueOf(expected);
+    }
+
+    public ExpectSObject andIt { get { return this; } }
+
+    public ExpectSObject shouldEqual(SObject actual) {
+        shouldEqual(actual, expectedMessage(actual,''));
+        return this;
+    }
+
+    public ExpectSObject shouldEqual(SObject actual, String message) {
+        return assertSObject(expected === actual, message);
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual) {
+        return shouldNotEqual(actual, expectedMessage(actual,' not'));
+    }
+
+    public ExpectSObject shouldNotEqual(SObject actual, String message) {
+        return assertSObject(expected !== actual, message);
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObject actual) {
+        return sObjectTypeEquals(actual, expectedMessage(actual.getSObjectType(),''));
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObject actual, String message) {
+        return sObjectTypeEquals(actual.getSObjectType(), message);
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObjectType objType) {
+        return sObjectTypeEquals(objType, expectedMessage(objType,''));
+    }
+
+    public ExpectSObject sObjectTypeEquals(SObjectType objType, String message) {
+        return assertSObject(expected.getSObjectType() == objType, message);
+    }
+
+    public ExpectSObject shouldFailToUpsert() {
+        return shouldFailToUpsertWithMessage('Upsert failed');
+    }
+
+    public ExpectSObject shouldFailToUpsertWithMessage(String message) {
```

Comment:
"Upsert failed" is the default start to every upsert failure message, which is why I had set it up like that - if you aren't testing for something specific (and there are many failure messages that can be tested against - STRING_TOO_LONG, DUPLICATE_VALUE, FIELD_INTEGERITY_EXCEPTION, etc ...) but just want to verify that regardless of the actual message, the action failed, then you can use the default shouldFailToUpsert() call.  

But yes, let's get the other changes merged and talk more about it in person!